src/core/doc_generators/filesystem.py

```
@@ -49,31 +49,34 @@ def __init__(self) -> None:
         self.files: dict[Path, str] = {}
         self.directories: set[Path] = set()
 
-    def write_text(self, path: Path, content: str, encoding: str = "utf-8") -> None:
+    def write_text(self, path: str | Path, content: str, encoding: str = "utf-8") -> None:
```

Ruff (0.14.0)

Unused method argument: `encoding`

(ARG002)